toolchain/parse/state.def

```diff
@@ -229,7 +229,7 @@ CARBON_PARSE_STATE(DeclarationNameAndParamsAfterImplicit)
 // If `Semi`:
 //   1. DeclarationScopeLoop
 // If `Var`:
-//   1. VarAsSemicolon
+//   1. VarAsDeclaration
```

Comment:
Please make sure `Returned` handling is documented.

Reply:
Done.